contributor-book/src/guides/adding-a-new-operation-to-burn.md

```diff
@@ -0,0 +1,103 @@
+# Adding a new operation to burn
+
+Let's discuss how one might go about adding new operators to burn, using the example of the recently added [pow operator](https://github.com/tracel-ai/burn/pull/1133/files) (as of 01/24/2024). In that PR, the following things took place (albeit not in this order)
+
+## Adding the Op to burn-tensor
+
```

Comment:
I would actually add an explanation for the 7 kinds of operations:

`base` => All tensor kinds should implement these operations (Reshape, into_data, etc.).

The implementation is in `burn-tensor/src/tensor/api/base.rs`.

`numeric` => All tensors that are numeric by nature should implement these operations (Add, Sub, Div, etc.).

The implementation is in `burn-tensor/src/tensor/api/numeric.rs`.

`float` => Tensor operations are only available for float tensors.

The implementation is in `burn-tensor/src/tensor/api/float.rs`.

`int` => Tensor operations are only available for int tensors.

The implementation is in `burn-tensor/src/tensor/api/int.rs`.

`bool` => Tensor operations are only available for bool tensors.

The implementation is in `burn-tensor/src/tensor/api/bool.rs`.

`module` => These should be exported as functions instead of methods on tensors.

The implementation is in `burn-tensor/src/tensor/module.rs` (Maybe refactor to `tensor/api/module.rs`).

```rust
/// Applies a 2D convolution (crate::ops::ModuleOps::conv2d).
pub fn conv2d<B>(
    x: Tensor<B, 4>,
    weight: Tensor<B, 4>,
    bias: Option<Tensor<B, 1>>,
    options: ConvOptions<2>,
) -> Tensor<B, 4>
where
    B: Backend,
{
    Tensor::new(B::conv2d(
        x.primitive,
        weight.primitive,
        bias.map(|b| b.primitive),
        options,
    ))
}
```

`activation` => These should also be exported as functions instead of methods on tensors.

The implementation is in `burn-tensor/src/tensor/activation/base.rs` (Maybe refactor to `tensor/api/activation.rs`).

Note that some activations are just a combination of backend operations and are not declared in `burn-tensor/src/tensor/ops/activation.rs`.

Reply:
Should this go into the guide for adding operators, or somewhere under the architecture directory(or within `architecture/tensor`, and then referenced from the guide?